azkaban-db/src/main/java/azkaban/db/MySQLDataSource.java

```diff
@@ -25,10 +25,7 @@
 
   private static final Logger logger = Logger.getLogger(MySQLDataSource.class);
 
-  private static volatile MySQLDataSource instance = null;
-
-  // TODO kunkun-tang: have guice inject working here
-  private MySQLDataSource(final String host, final int port, final String dbName,
+  public MySQLDataSource(final String host, final int port, final String dbName,
```

Comment:
This class is indirectly created by Guice already. Right? 
And it doesn't require Guice's assistance to create an instance. So I think what you have is fine.